firmware/targets/f7/furi_hal/furi_hal_speaker.c

```diff
@@ -54,6 +88,8 @@ static inline uint32_t furi_hal_speaker_calculate_compare(float volume) {
 }
 
 void furi_hal_speaker_start(float frequency, float volume) {
+    furi_check(furi_mutex_get_owner(furi_hal_speaker_bus_mutex) == furi_thread_get_current_id());
```

Comment:
What about interrupts?

Reply:
also you already have furi_hal_speaker_is_mine which can be reused here.